src/main/java/org/springframework/data/repository/config/RepositoryConfigurationExtensionSupport.java

```diff
@@ -106,7 +109,13 @@ public String getDefaultNamedQueryLocation() {
 	}
 
 	public void registerBeansForRoot(BeanDefinitionRegistry registry,
-			RepositoryConfigurationSource configurationSource) {}
+			RepositoryConfigurationSource configurationSource) {
+
+		// A proxy RepositoryMethodContext for dependency injection
+		registerIfNotAlreadyRegistered(
```

Comment:
Instance supplier is not supported in AOT and will raise `org.springframework.beans.factory.aot.AotBeanProcessingException`.